### ISSUE TITLE
Parentheses in with statment is erroneously identified as python 3.10 parenthesized context manager

### ISSUE THREAD

**Issue by alejo90** [microsoft/pyright#1552]
**Describe the bug**
Using a parenthesized expression in a `with` statement is erroneously identified (and marked as an error) as an upcoming python 3.10 feature called [parenthesized context managers](https://docs.python.org/3.10/whatsnew/3.10.html#parenthesized-context-managers) by pyright 1.1.115. 

**To Reproduce**
Use pyright 1.1.115 and python 3.9 and enter the snippet below in a python file.

**Expected behavior**
Pyright would distinguish between multiple context managers wrapped in parentheses (python 3.10's parenthesized context managers) and a single context manager containing and parenthesized expression and not mark the latter as an error when using python < 3.10.

**Screenshots or Code**
```python
from pathlib import Path

root_path = Path("foo")

with (root_path / "bar.py").open(mode="r") as f:
    # do something with f
    ...
```
![image](https://user-images.githubusercontent.com/1158694/109440203-f5577f80-79f6-11eb-8752-f4588e7a61d1.png)

**VS Code extension or command-line**
I'm running pyright 1.1.115 as a command line tool via coc-pyright in neovim. The issue also started appearing in vscode with Pylance 2021.2.4. It wasn't happening with the previous version.

**Comment by erictraut** [comment]
Thanks for the bug report. We already noticed this problem, and a fix is in place for the next release.

**Comment by alejo90** [comment]
Wow, that was a fast response. Thanks!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.116, which I just published. It will also be in the next release of pylance.
